Submitty/Submitty
# [Bugfix:Submission] Refactor Notebook Autosave

### What is the current behavior?
closes #5351 
Autosaving notebook answers currently relies on numbered id's for each question which is what leads to the issue above.

### What is the new behavior?
Autosaving now uses the question filename (determined in the config.json) as a key and since each question is required to have a unique filename, there are no issues.

### Other information?
To Test:
1. Test a lot because this is a total refactor of how autosave works
2. Fill out a notebook gradeable
3. Add a question to the config, reupload the config
4. refresh the notebook
5. If on master, see that the answers after the new question you added are wrong
6. if on new branch, see that the answers are correct
